### PR TITLE
normalize numeric and date values 

### LINKED ISSUE #251: MAP: normalize numeric and date values
As an ARLAS maintainer, I want to normalize some numeric or date fields value. The normalized fields must be configurable.

Moreover, I want to be able to normalize the fields values locally on the current extent OR globally considering the current filters.

Finally I also want to be able to normalize some fields values (locally or globally) per a given key : For instance I want to normalize the speed of boats for each boat id

- [x] Configure fields to normalize
- [x] Normalize fields locally
- [x] Normalize fields locally per key
- [x] Normalize fields gloabally per key

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**
Fix #251

### CODE REVIEW

File: src/contributors/MapContributor.ts [hunk 1]
```diff
@@ -216,6 +243,22 @@ export class MapContributor extends Contributor {
                     this.aggregationField = collection.params.centroid_path;
                 }
             });
+
+        if (this.includeFeaturesFields) {
+            this.allIncludedFeatures = new Set();
+            this.includeFeaturesFields.forEach(f => this.allIncludedFeatures.add(f));
+            this.includeFeaturesFieldsSet = new Set(this.includeFeaturesFields);
+            if (this.normalizationFields) {
+                this.normalizationFields.forEach(f => {
```
Comment (mbarbet) [inline src/contributors/MapContributor.ts]: Could you use filter instead of for each ?

File: src/contributors/MapContributor.ts [hunk 2]
```diff
@@ -1189,6 +1236,176 @@ export class MapContributor extends Contributor {
...
+                        {
+                            collect_fct: Metric.CollectFctEnum.MIN,
+                            collect_field: f.on
+                        },
+                        {
+                            collect_fct: Metric.CollectFctEnum.MAX,
+                            collect_field: f.on
+                        }
+                    ],
+                    size: '10000'
```
Comment (mbarbet) [inline src/contributors/MapContributor.ts]: Dangereous for performance ?

File: src/contributors/MapContributor.ts (hunk 1 above)
Comment (MohamedHamouGisaia) [reply src/contributors/MapContributor.ts]: done

File: src/contributors/MapContributor.ts (hunk 2 above)
Comment (MohamedHamouGisaia) [reply src/contributors/MapContributor.ts]: A configuration variable is added

**Pull request merged**